### DISCUSSION TITLE
HmIP-RFUSB-TK und DualCoPro Firmware

### DISCUSSION THREAD

**Discussion by jp112sdl** [OpenCCU/OpenCCU#2155] [🇩🇪 Generelle Diskussionen]
Ich wollt mal fragen, an welcher Stelle das Update eines HmIP-RFUSB-TK auf die [dualcopro_update_blhmip-4.4.18.eq3](https://github.com/eq-3/occu/blob/master/firmware/HmIP-RFUSB/dualcopro_update_blhmip-4.4.18.eq3) Firmware fehlschlägt.

Man liest ja überall, dass sich der HmIP-RFUSB-TK Stick nicht für BidCos und HmIP flashen lässt.

Aber was ist die konkrete Fehlermeldung?

**Comment by jens-maus** [comment]
Das ist ne gute Frage. Meines Wissens verweigert das HmIP-RFUSB-TK einfach die Annahme der Firmware weil es eine andere Product-ID hat. So zumindest meine Infos dazu. Aber müsste man mal probieren, hast recht.

**Reply by jp112sdl** [reply]
Fällt dir jemand ein, der so ein -TK besitzt?

**Reply by jens-maus** [reply]
Ich müsste noch so eins irgendwo auf dem Tisch haben. Teste ich mal und geb bescheid was bei rauskommt.

**Reply by jens-maus** [reply]
Wie gesagt, irgendwas scheint da noch abgeprüft zu werden das der `HmIP-RFUSB-TK` das firmware-update des nicht-TK nicht annimmt:

```
# /opt/java/bin/java -Dos.arch=x86_64 -Dgnu.io.rxtx.SerialPorts=/dev/raw-uart -jar /op
t/HmIP/hmip-copro-update.jar -p /dev/raw-uart -o -f /firmware/HmIP-RFUSB/dualcopro_update_blhmip-4.4.18.eq3 
[INFO] Homematic IP coprocessor update tool V1.0.10
[DEBUG] Start bootloader ...
[DEBUG] Bootloader running
[DEBUG] Request bootloader version ...
[INFO] SGTIN = 3014F5AC9400041DXXXXXXXX
[INFO] Performing update with firmware file '/firmware/HmIP-RFUSB/dualcopro_update_blhmip-4.4.18.eq3' ...
[DEBUG] Start application ...
[DEBUG] Application 'DualCoPro_App' running
[DEBUG] Request application version ...
[DEBUG] Request application version ...
[INFO] Updating coprocessor from version 4.2.14 to version 4.4.18
[DEBUG] Start bootloader ...
[DEBUG] Bootloader running
[INFO] Entering bootloader ...
[DEBUG] Start bootloader ...
[DEBUG] Already in Bootloader.
[DEBUG] Request bootloader version ...
[INFO] Bootloader version: 1.0.24
[DEBUG] Send update frames ...
[ERROR] No update performed.
[INFO] Exiting update tool.
```

Danach ist dann der Stick weiterhin auf der 4.2.14 Version:
```
# detect_radio_module /dev/raw-uart
HMIP-RFUSB-TK 1DXXXXXXXX 3014F5AC9400041DXXXXXXXX 0xFFE8XX 0xBB54XX 4.2.14
```

Ich vermute irgendwas im bootloader prüft wohl ob das Firmware-Update auch zu dem Stick passt.

**Reply by jp112sdl** [reply]
```
[DEBUG] Send update frames ...
[ERROR] No update performed.
```

Kommen die Ausgaben unverzüglich nacheinander oder vergehen dazwischen ein paar Sekunden?

**Reply by jens-maus** [reply]
Ich würde sagen unverzüglich.

**Reply by jens-maus** [reply]
Siehe:

```
# /opt/java/bin/java -Dos.arch=x86_64 -Dgnu.io.rxtx.SerialPorts=/dev/raw-uart -jar /op
t/HmIP/hmip-copro-update.jar -p /dev/raw-uart -o -f /firmware/HmIP-RFUSB/dualcopro_update_blhmip-4.4.18.eq3 | t
s '[%H:%M:%S]'
[09:42:09] [INFO] Homematic IP coprocessor update tool V1.0.10
RXTX Warning:  Removing stale lock file. /var/lock/LCK..raw-uart
[09:42:09] [INFO] SGTIN = 3014F5AC9400041DXXXXXXXX
[09:42:09] [INFO] Performing update with firmware file '/firmware/HmIP-RFUSB/dualcopro_update_blhmip-4.4.18.eq3' ...
[09:42:09] [DEBUG] Start application ...
[09:42:10] [DEBUG] Application 'DualCoPro_App' running
[09:42:10] [DEBUG] Request application version ...
[09:42:10] [DEBUG] Request application version ...
[09:42:10] [INFO] Updating coprocessor from version 4.2.14 to version 4.4.18
[09:42:10] [DEBUG] Start bootloader ...
[09:42:10] [DEBUG] Bootloader running
[09:42:10] [INFO] Entering bootloader ...
[09:42:10] [DEBUG] Start bootloader ...
[09:42:10] [DEBUG] Already in Bootloader.
[09:42:10] [DEBUG] Request bootloader version ...
[09:42:10] [INFO] Bootloader version: 1.0.24
[09:42:10] [DEBUG] Send update frames ...
[09:42:10] [ERROR] No update performed.
[09:42:10] [INFO] Exiting update tool.
```

**Reply by jp112sdl** [reply]
Hmm, das ist ja doof. Aber die HmIP-only Firmware lässt sich einspielen?

**Reply by jens-maus** [reply]
Welche? Die 2.8.6?!? Hmm müsste ich probieren.

**Reply by jp112sdl** [reply]
Ja genau, [`hmip_coprocessor_update.eq3`](https://github.com/eq-3/occu/blob/master/firmware/HmIP-RFUSB/hmip_coprocessor_update.eq3)

**Reply by jens-maus** [reply]
"Same same, but different":

```
# /opt/java/bin/java -Dos.arch=x86_64 -Dgnu.io.rxtx.SerialPorts=/dev/raw-uart -jar /op
t/HmIP/hmip-copro-update.jar -p /dev/raw-uart -o -f /firmware/HmIP-RFUSB/hmip_coprocessor_update-2.8.6.eq3 | ts
 '[%H:%M:%S]'
[13:54:58] [INFO] Homematic IP coprocessor update tool V1.0.10
RXTX Warning:  Removing stale lock file. /var/lock/LCK..raw-uart
[13:54:58] [INFO] SGTIN = 3014F5AC9400041DXXXXXXX
[13:54:58] [INFO] Performing update with firmware file '/firmware/HmIP-RFUSB/hmip_coprocessor_update-2.8.6.eq3' ...
[13:54:58] [DEBUG] Start application ...
[13:54:59] [DEBUG] Application 'DualCoPro_App' running
[13:54:59] [DEBUG] Request application version ...
[13:54:59] [DEBUG] Request application version ...
[13:54:59] [INFO] Updating coprocessor from version 4.2.14 to version 2.8.6
[13:54:59] [DEBUG] Start bootloader ...
[13:54:59] [DEBUG] Bootloader running
[13:54:59] [INFO] Entering bootloader ...
[13:54:59] [DEBUG] Start bootloader ...
[13:54:59] [DEBUG] Already in Bootloader.
[13:54:59] [DEBUG] Request bootloader version ...
[13:54:59] [INFO] Bootloader version: 1.0.24
[13:54:59] [DEBUG] Send update frames ...
[13:55:00] [ERROR] No update performed.
[13:55:00] [INFO] Exiting update tool.
```

Mit dem Einsatz eines `HmIP-RFUSB-TK` ist man also ganz schön abgehängt. Und auf nachfrage bei eQ3 sind da auch keine weiteren Firmwareupdates mehr dazu geplant. Die Telekom als OEM Partner definiert wohl was da kommt und was nicht und man will denen jetzt nicht in die Suppe spucken wenn man dafür die 4.4er freigibt und dann damit Leute in der Telekom-Cloud damit auftauchen und Probleme haben. Deshalb bleibt es wohl bis auf weiteres dabei das nur ein `HmIP-RFUSB` wirklich sinnvoll einsetzbar ist... oder aber du findest natürlich einen Weg die Firmware-Update-Routinen auszutricksen ;-)

**Reply by jp112sdl** [reply]
Ich glaub der Aufwand lohnt sich nicht.
Interessant ist halt schon, dass sowohl Company Code der SGTIN (3014-...) bei HmIP-RFUSB und HmIP-RFSUB-TK identisch sind und auch die Device ID (271), die in der SGTIN steckt (bestimmte Bit-Anteile aus `041D`).

Einzig der "manufacturer code" scheint direkt im Gerät zu stecken.
Und daran wird wohl auch das Update scheitern. Der -TK Stick verweigert die Annahme der Firmware-Pakete.
Man könnte jetzt noch in der `hmip-copro-update.jar` rumfummeln und das Übertragen der Pakete auch nach nicht erfolgreichem Ack weiterführen. 
Aber ich glaub, das lohnt sich wirklich nicht.

Btw: Was passiert eigentlich, wenn man den Stick mit `eq3configcmd` füttern will?
https://github.com/jens-maus/RaspberryMatic/blob/efd47b639b0d639da7828296c269d0ab9360fcdb/buildroot-external/overlay/base/etc/init.d/S48UpdateRFHardware#L72

**Reply by jens-maus** [reply]
> Btw: Was passiert eigentlich, wenn man den Stick mit `eq3configcmd` füttern will?

Da kommen auch nur fehler:

```
# eq3configcmd update-coprocessor -p /dev/raw-uart -u -f -t HM-MOD-UART -c -d /firmware/HmIP-RFUSB
2023/01/12 15:33:14.423 <Info> CCU2CommControllerMod::sendSystemCommand(): failed
2023/01/12 15:33:14.424 <Error> CCU2CommControllerMod::performIdentify(): Unable to determine coprocessor state.
2023/01/12 15:33:17.982 <Info> CCU2CommControllerMod::sendSystemCommand(): failed
2023/01/12 15:33:22.985 <Error> CoprocessorUpdate::startBootloader():Could not start Coprocessor bootloader.

2023/01/12 15:33:23.577 <Info> CCU2CommControllerMod::sendSystemCommand(): failed
```

**Reply by jp112sdl** [reply]
Narf... na gut, dann wär das auch alles durchgetestet
